### PR TITLE
Awesome Books with ES6 modifications

### PR DESCRIPTION
**Pull request by charlyeneh**
Features Implemented:
- I used ES6 modules to write modular JavaScript.
- The use of npm as a software packages system.
- Use of luxon library to implement date.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "hint": "^6.1.3",
+    "stylelint": "^14.5.0",
+    "stylelint-config-standard": "^25.0.0",
+    "stylelint-csstree-validator": "^2.0.0",
+    "stylelint-scss": "^4.1.0"
+  },
+  "dependencies": {
+    "luxon": "^2.3.0"
+  }
+}
```
Comment (elmar8287) [inline package.json:34]: - Luxon installed 👍

File: index.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+const book3 = new Book('title3', 'author3', '003');
+
+books = [book1, book2, book3];
+
+// add event listner to newly added book remove button
+let removeBtn = document.querySelectorAll('.remove');
+
+function removeBook(ref) {
+  const result = books.filter((value) => value.id !== ref);
+  books = result;
```
Comment (elmar8287) [inline index.js]: - [ ] Please, consider using arrow function syntax, as required in the project (use ES6 syntax). For example, here you can go on with 
```
const removeBook = (ref) => {
....
```

- [ ] Also, will be better to create an additional module for removing the books method, then import it here in `index.js` file

File: index.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  addToLocalStorage(books);
+}
+
+function addEvents() {
```
Comment (elmar8287) [inline index.js]: - [ ] Please, consider using arrow function syntax, as required in the project (use ES6 syntax). For example, here you can go on with 
```
const addEvents = () => {
....
```

File: index.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  removeBtn.forEach((element) => {
+    element.addEventListener('click', (e) => {
+      const ref = e.target.id;
+      removeBook(ref);
+      e.target.parentElement.remove();
+    });
+  });
+}
+
+// populate dom with the list
```
Comment (elmar8287) [inline index.js:32]: - You can make your code cleaner by removing all comments. Your code is talking about everything itself 👍

File: index.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+const addBtn = document.querySelector('#add-btn');
+function showBooks(list) {
```
Comment (elmar8287) [inline index.js]: - [ ] Please, consider using arrow function syntax, as required in the project (use ES6 syntax). For example, here you can go on with 
```
const showBooks = (list) => {
....
```

File: index.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  }
+  removeBtn = document.querySelectorAll('.remove');
+  addEvents();
+}
+const listClass = document.getElementById('List');
+const addNewClass = document.getElementById('Add_new');
+const contactClass = document.getElementById('Contact');
+
+// add book to the list
+function addBook() {
```
Comment (elmar8287) [inline index.js]: - [ ] Please, consider using arrow function syntax, as required in the project (use ES6 syntax). For example, here you can go on with 
```
const addBook = () => {
....
```

- [ ] Also, will be better to create an additional module for adding the books method, then import it here in the `index.js` file

File: modules/localStorage.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+// add data to local storage
+
+function addToLocalStorage(data) {
```
Comment (elmar8287) [inline modules/localStorage.js]: - [ ] Please, consider using arrow function syntax, as required in the project (use ES6 syntax). For example, here you can go on with 
```
const addToLocalStorage = (data) => {
....
```

- [ ] Also, will be better to create an additional module for storing data, then import it here in the `index.js` file

**Review by elmar8287 — CHANGES_REQUESTED** [review]
Hi @charlyeneh, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  Check the comments under the review.

## Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by charlyeneh** [comment]
Hi Reviewer,
Thank you for the corrections.

I have changed all functions to ES6 arrow function syntax as suggested.

Regards!

**Review by rustam-tolipov — APPROVED** [review]
## Hi @charlyeneh 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by elmar8287** [comment]
> Hi Reviewer, Thank you for the corrections.
> 
> I have changed all functions to ES6 arrow function syntax as suggested.
> 
> Regards!

@charlyeneh Perfect! 👍

**Pull request merged**